lua/luasnip/config.lua

```diff
@@ -24,6 +24,7 @@ local defaults = {
 	region_check_events = "User None",
 	delete_check_events = "User None",
 	store_selection_keys = nil, -- Supossed to be the same as the expand shortcut
+	store_meta_data = false, -- For each snippet, eg. source path.
```

Comment:
I think I'd like this to be a table, which can accept keys to determine what is stored. Most capable would be just a function which receives all available data and creates the metadata for the snippet, but I think this is overdoing it, since we probably don't need much flexibility there.

Still, making this a table would somewhat future-proof it, like

```lua
metadata = {sourcefile=true, line=true, custom = somefn}
```

Reply:
I think using a subtable is good idea. I don't understand your reasoning about the function.

Reply:
Nice, thanks!
My idea with the function was to allow complete customization of what is put into the metadata, essentially by calling a function like

```lua
function loader_metadata(filename, line)
	return {source=filename, source_line = line}
end
```
in the loaders.

Since there isn't this much customization possible (I think), letting the user set some keys for what exactly should be stored in the metadata-table seems like a more straightforward, and easier to use simplification.

Reply:
Ill try to making it into a func and see how it feels

Reply:
Now, this is the only thing left to do.

Reply:
I believe that we can have default as a simple `false` and then check `if func` and then run the user meta func, otherwise just default to nothing.

Reply:
and ig also if user sets store meta == true, then store everything?

Reply:
How would one get the source line of the snippet in this function? Is there something builtin to luasnip already?

Reply:
i pushed a basic version

Reply:
> How would one get the source line of the snippet in this function? Is there something builtin to luasnip already?

Oh, that's very hard to do for lua and vscode, I used it because I wanted an example for why more flexibility might be good.
I think we should not determine all lines up-front (except if it's trivial, like in the snipmate-loader), since only a tiny subset of this information will be useful, and it should, I think, be calculated on-demand

Reply:
Maybe not very hard, non-trivial

Reply:
> Maybe not very hard, non-trivial

Aight, for now my feed keys thingy works good nuff

Reply:
Right! heuristics like that should do it for 99%